CMakeLists.txt

```diff
@@ -216,7 +249,18 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/lint.cmake)
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Install.cmake)
 
-add_library(DALI::dali ALIAS dali)
-add_library(DALI::dali_core ALIAS dali_core)
-add_library(DALI::dali_kernels ALIAS dali_kernels)
-add_library(DALI::dali_operators ALIAS dali_operators)
+if (BUILD_DALI_CORE)
+  add_library(DALI::dali_core ALIAS dali_core)
+endif()
+
+if (BUILD_DALI_KERNELS)
+  add_library(DALI::dali_kernels ALIAS dali_kernels)
+endif()
+
+if (BUILD_DALI_PIPELINE)
+  add_library(DALI::dali ALIAS dali)
+endif()
+
+if (BUILD_DALI_OPERATORS)
+  add_library(DALI::dali_operators ALIAS dali_operators)
+endif()
```

Comment:
```
if (BUILD_DALI_CORE)
  add_library(DALI::dali_core ALIAS dali_core)
  if (BUILD_DALI_KERNELS)
    add_library(DALI::dali_kernels ALIAS dali_kernels)
    if (BUILD_DALI_PIPELINE)
      add_library(DALI::dali ALIAS dali)
      if (BUILD_DALI_OPERATORS)
       add_library(DALI::dali_operators ALIAS dali_operators)
      endif()
    endif()
  endif()
endif()
```
?

Reply:
I am thinking to make at least core to always build